src/main/java/VASSAL/counters/PropertyExporter.java

```diff
@@ -0,0 +1,20 @@
+package VASSAL.counters;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import VASSAL.build.module.properties.PropertyNameSource;
+import VASSAL.build.module.properties.PropertySource;
+
+public interface PropertyExporter extends PropertyNameSource, PropertySource {
+
+  default Map<String, Object> getProperties() {
+    final List<String> propertyNames = getPropertyNames();
+    java.util.Map<String, Object> result = new HashMap<>();
+    for (String propertyName : propertyNames) {
+      result.put(propertyName, getLocalizedProperty(propertyName));
+    }
+    return result;
```

Comment:
The list of propertyNames will never change once a GamePiece has been created. It is also relatively expensive to retrieve the list of names. Is it possibly to cache the propertyNames list in a static variable inside an interface default method implementation?

Reply:
Could cache them in a `static final List<String>` but that would be a single List for all PropertyExporter objects in the JVM. Besides, remember what Donald Knuth said about premature optimization, once we optimize away the `PieceCloner.clone()` we can do some new measurements and if we see that `getPropertyNames()` is a major performance hog we can still see about caching the results. Some of its implementations are basically `return new ArrayList() { A, B, C }`, they look easy enough for the JVMs to optimize them during runtime if they are called often enough. Modern JVMs do wonders with small methods, they can turn them to native code on the fly.

Reply:
Sounds good to me.